TravelAgency-spring-mvc/src/main/java/cz/muni/fi/travelAgency/controllers/CustomerController.java

```diff
@@ -63,7 +63,14 @@ public String submitCreate(@Valid @ModelAttribute("customerCreate") CustomerCrea
         return "redirect:" + uriBuilder.path("/").build().encode().toUriString();
     }
 
-    @RequestMapping(value = "/edit", method = RequestMethod.GET)
+
+    @RequestMapping(value = "detail/{id}", method = RequestMethod.GET)
+    public String detail(@PathVariable long id, Model model) {
+        model.addAttribute("customer", customerFacade.findCustomerById(id));
+        return "admin/customer/detail";
```

Comment:
This is a tricky one. You are sending any user that edited his account to a page he will not have access to. I suggest moving the admin/customer/detail to the customers scope (the page and rather adjust admin controllers to access the page customer/detail)